tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -142,7 +144,100 @@ def test_brand_string(test_microvm_with_api, network_config):
 @pytest.mark.skipif(
     PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
 )
-@pytest.mark.parametrize("cpu_template", ["T2", "C3"])
+@pytest.mark.skipif(
+    cpuid_utils.get_cpu_vendor() != cpuid_utils.CpuVendor.INTEL,
+    reason="CPU templates are only available on Intel.",
+)
+@pytest.mark.skipif(
+    utils.get_kernel_version(level=1) not in ["5.10", "4.14"],
+    reason="Supported kernels are 5.10 and 4.14",
+)
+@pytest.mark.parametrize("cpu_template", ["T2S"])
+@pytest.mark.timeout(500)
+@pytest.mark.nonci
+def test_cpu_rdmsr(test_microvm_with_api_and_msrtools, network_config, cpu_template):
```

Comment:
I d rather just update the current test_microvm_with_api image. The fixture thing is a very debated and limited design (it does not allow for multiple guest kernels to be tested) so I would try to be reserved as to adding new ones.

Reply:
I'm reluctant to update the current image since doing that changes packages and can impact other tests in the CI so I want to keep the blast radius small.
We can switch this test to use `ArtifactSet` instead though.

Reply:
Switched to TestMatrix and added 5.10 Guest.